packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -773,6 +773,13 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         case tt.bracketL:
           return this.tsParseTupleType();
         case tt.parenL:
+          if (!this.options.createParenthesizedExpressions) {
+            this.expect(tt.parenL);
+            const type = this.tsParseType();
+            this.expect(tt.parenR);
+            return type;
+          }
```

Comment:
Can you use `addExtra` to add info about parenths, like we do here?

https://github.com/babel/babel/blob/f9d4b799b74d4cc381679a3124f2c8c58745f4c0/packages/babel-parser/src/parser/expression.js#L1335-L1336